plugins/gemalto.c

```diff
@@ -1381,54 +1447,21 @@ static void gemalto_remove(struct ofono_modem *modem)
 		g_at_chat_cancel_all(data->app);
 		g_at_chat_unregister_all(data->app);
 		g_at_chat_unref(data->app);
+		if (data->mdm == data->app) {
+		  data->mdm = NULL;
+		}
 		data->app = NULL;
 	}
 
 	if (data->mdm) {
```

Comment:
please use the boolean here like:
   if(data->use_mdm_for_app) data->mdm=NULL;
between the two if's.